File: 2.py
if __name__ == "__main__":
    f = [0, 1]
    even = []
    i = 1
    while f[i] <= 4000000:
        
        if f[i] % 2 == 0:
            even.append(f[i])

        i += 1
        f.append(f[i-1] + f[i-2])
        
    print(sum(even))

File: 2.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    vector<int> fib = {1, 1};
    int sum = 0;

    while (fib.back() < 4'000'000) {
        int n = fib.size();
        fib.push_back(fib[n - 1] + fib[n - 2]);
       
        if (fib.back() % 2 == 0) 
            sum += fib.back();
    }

    cout << "Sum of even Fibonacci numbers: " << sum << endl;
    return 0;
}
